diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/krb4-1.2.2/lib/acl/Makefile.am src/krb4-1.2.2/lib/acl/Makefile.am
--- origsrc/krb4-1.2.2/lib/acl/Makefile.am	2002-09-09 08:24:31.000000000 -0500
+++ src/krb4-1.2.2/lib/acl/Makefile.am	2007-07-04 20:24:27.828125000 -0500
@@ -9,7 +9,7 @@
 build_HEADERZ = $(include_HEADERS)
 
 lib_LTLIBRARIES = libacl.la
-libacl_la_LDFLAGS = -version-info 1:0:0
+libacl_la_LDFLAGS = -version-info 1:0:0 -no-undefined
 libacl_la_LIBADD  = ../krb/libkrb.la
 
 libacl_la_SOURCES = acl_files.c
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/krb4-1.2.2/lib/com_err/Makefile.am src/krb4-1.2.2/lib/com_err/Makefile.am
--- origsrc/krb4-1.2.2/lib/com_err/Makefile.am	2002-03-10 17:52:41.000000000 -0600
+++ src/krb4-1.2.2/lib/com_err/Makefile.am	2007-07-04 20:24:27.828125000 -0500
@@ -5,7 +5,7 @@
 YFLAGS = -d
 
 lib_LTLIBRARIES = libcom_err.la
-libcom_err_la_LDFLAGS = -version-info 2:1:1
+libcom_err_la_LDFLAGS = -version-info 2:1:1 -no-undefined
 
 bin_PROGRAMS = compile_et
 
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/krb4-1.2.2/lib/des/Makefile.am src/krb4-1.2.2/lib/des/Makefile.am
--- origsrc/krb4-1.2.2/lib/des/Makefile.am	2001-08-03 22:10:23.000000000 -0500
+++ src/krb4-1.2.2/lib/des/Makefile.am	2007-07-04 20:24:27.843750000 -0500
@@ -3,7 +3,7 @@
 include $(top_srcdir)/Makefile.am.common
 
 lib_LTLIBRARIES = libdes.la
-libdes_la_LDFLAGS = -version-info 3:1:0
+libdes_la_LDFLAGS = -version-info 3:1:0 -no-undefined
 
 include_HEADERS = des.h md4.h md5.h sha.h rc4.h
 
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/krb4-1.2.2/lib/des/rnd_keys.c src/krb4-1.2.2/lib/des/rnd_keys.c
--- origsrc/krb4-1.2.2/lib/des/rnd_keys.c	2001-08-21 10:32:07.000000000 -0500
+++ src/krb4-1.2.2/lib/des/rnd_keys.c	2007-07-04 20:24:27.843750000 -0500
@@ -163,7 +163,7 @@
 static volatile int igdata;	/* Index into global data */
 static int gsize;
 
-#if !defined(WIN32) && !defined(__EMX__) && !defined(__OS2__) && !defined(__CYGWIN32__)
+#if !defined(WIN32) && !defined(__EMX__) && !defined(__OS2__)
 /* Visual C++ 4.0 (Windows95/NT) */
 
 static
@@ -203,7 +203,7 @@
 }
 #endif
 
-#if !defined(WIN32) && !defined(__EMX__) && !defined(__OS2__) && !defined(__CYGWIN32__)
+#if !defined(WIN32) && !defined(__EMX__) && !defined(__OS2__)
 
 #ifndef HAVE_SETITIMER
 static void
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/krb4-1.2.2/lib/kadm/Makefile.am src/krb4-1.2.2/lib/kadm/Makefile.am
--- origsrc/krb4-1.2.2/lib/kadm/Makefile.am	2002-09-09 14:17:04.000000000 -0500
+++ src/krb4-1.2.2/lib/kadm/Makefile.am	2007-07-04 20:24:27.859375000 -0500
@@ -9,7 +9,7 @@
 build_HEADERZ = $(include_HEADERS)
 
 lib_LTLIBRARIES = libkadm.la
-libkadm_la_LDFLAGS = -version-info 1:11:0
+libkadm_la_LDFLAGS = -version-info 1:11:0 -no-undefined
 libkadm_la_LIBADD  = ../krb/libkrb.la $(LIB_des) $(LIB_com_err)
 
 libkadm_la_SOURCES =				\
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/krb4-1.2.2/lib/kafs/Makefile.am src/krb4-1.2.2/lib/kafs/Makefile.am
--- origsrc/krb4-1.2.2/lib/kafs/Makefile.am	2002-08-19 10:08:37.000000000 -0500
+++ src/krb4-1.2.2/lib/kafs/Makefile.am	2007-07-04 20:24:27.859375000 -0500
@@ -48,7 +48,7 @@
 endif # KRB5
 
 lib_LTLIBRARIES = $(AFSLIBS)
-libkafs_la_LDFLAGS = -version-info 3:4:3
+libkafs_la_LDFLAGS = -version-info 3:4:3 -no-undefined
 foodir = $(libdir)
 foo_DATA = $(AFS_EXTRA_LIBS)
 # EXTRA_DATA = afslib.so
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/krb4-1.2.2/lib/kdb/Makefile.am src/krb4-1.2.2/lib/kdb/Makefile.am
--- origsrc/krb4-1.2.2/lib/kdb/Makefile.am	2002-09-09 14:17:05.000000000 -0500
+++ src/krb4-1.2.2/lib/kdb/Makefile.am	2007-07-04 20:24:28.234375000 -0500
@@ -9,8 +9,8 @@
 build_HEADERZ = $(include_HEADERS)
 
 lib_LTLIBRARIES = libkdb.la
-libkdb_la_LDFLAGS = -version-info 1:11:0
-libkdb_la_LIBADD  = ../krb/libkrb.la $(LIB_des)
+libkdb_la_LDFLAGS = -version-info 1:11:0 -no-undefined
+libkdb_la_LIBADD  = ../krb/libkrb.la $(LIB_des) $(DBLIB)
 
 if HAVE_DB3
 ndbm_wrap = ndbm_wrap.c ndbm_wrap.h
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/krb4-1.2.2/lib/krb/Makefile.am src/krb4-1.2.2/lib/krb/Makefile.am
--- origsrc/krb4-1.2.2/lib/krb/Makefile.am	2002-09-09 13:42:38.000000000 -0500
+++ src/krb4-1.2.2/lib/krb/Makefile.am	2007-07-04 22:18:25.593750000 -0500
@@ -23,9 +23,9 @@
 
 
 lib_LTLIBRARIES = libkrb.la
-libkrb_la_LDFLAGS = -version-info 2:1:1
+libkrb_la_LDFLAGS = -version-info 2:1:1 -no-undefined
 
-libkrb_la_LIBADD = $(LIB_des)
+libkrb_la_LIBADD = $(LIB_des) $(LIB_crypt)
 
 if DES_QUAD_CKSUM_BROKEN
 des_quad_cksum = quad_cksum.c
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/krb4-1.2.2/lib/otp/Makefile.am src/krb4-1.2.2/lib/otp/Makefile.am
--- origsrc/krb4-1.2.2/lib/otp/Makefile.am	2002-08-13 09:02:54.000000000 -0500
+++ src/krb4-1.2.2/lib/otp/Makefile.am	2007-07-04 20:24:28.250000000 -0500
@@ -13,7 +13,7 @@
 include_HEADERS = otp.h
 
 lib_LTLIBRARIES = libotp.la
-libotp_la_LDFLAGS = -version-info 1:3:1
+libotp_la_LDFLAGS = -version-info 1:3:1 -no-undefined
 libotp_la_LIBADD  = $(LIB_des) $(LIB_roken) $(LIB_NDBM)
 
 if HAVE_DB3
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/krb4-1.2.2/lib/roken/Makefile.am src/krb4-1.2.2/lib/roken/Makefile.am
--- origsrc/krb4-1.2.2/lib/roken/Makefile.am	2002-09-09 14:17:01.000000000 -0500
+++ src/krb4-1.2.2/lib/roken/Makefile.am	2007-07-04 20:24:28.250000000 -0500
@@ -7,7 +7,7 @@
 CLEANFILES = roken.h make-roken.c $(XHEADERS)
 
 lib_LTLIBRARIES = libroken.la
-libroken_la_LDFLAGS = -version-info 16:0:0
+libroken_la_LDFLAGS = -version-info 16:0:0 -no-undefined
 
 noinst_PROGRAMS = make-roken snprintf-test
 
@@ -162,7 +162,7 @@
 
 EXTRA_DIST = roken.awk roken.h.in
 
-libroken_la_LIBADD = @LTLIBOBJS@ $(DBLIB)
+libroken_la_LIBADD = @LTLIBOBJS@ $(DBLIB) $(LIB_crypt)
 
 $(LTLIBOBJS) $(libroken_la_OBJECTS): $(include_HEADERS) roken.h $(XHEADERS)
 
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/krb4-1.2.2/lib/sl/Makefile.am src/krb4-1.2.2/lib/sl/Makefile.am
--- origsrc/krb4-1.2.2/lib/sl/Makefile.am	2002-08-13 08:48:17.000000000 -0500
+++ src/krb4-1.2.2/lib/sl/Makefile.am	2007-07-04 20:24:28.281250000 -0500
@@ -13,8 +13,8 @@
 include_HEADERS = sl.h
 
 lib_LTLIBRARIES = libsl.la libss.la
-libsl_la_LDFLAGS = -version-info 1:2:1
-libss_la_LDFLAGS = -version-info 1:4:1
+libsl_la_LDFLAGS = -version-info 1:2:1 -no-undefined
+libss_la_LDFLAGS = -version-info 1:4:1 -no-undefined
 
 libsl_la_LIBADD = @LIB_readline@
 libss_la_LIBADD = @LIB_readline@ @LIB_com_err@
